examples/utils.py

```diff
@@ -0,0 +1,149 @@
+"""Utilities for yapapi example scripts."""
+from collections import defaultdict
+from itertools import count
+import logging
+import time
+from typing import Any, Callable, Dict, Iterator, List, Optional, Set
+
+from yapapi.runner.events import Event, EventType, event_type_to_string
+
+
+class SummaryLogger:
+    """Aggregates information from computation events to provide a high-level description."""
+
+    logger = logging.getLogger("yapapi.summary")
+
+    # Generates subsequent numbers, for use in generated provider names
+    numbers: Iterator[int]
+
+    # Start time of the computation
+    start_time: float
+
+    # Set of confirmed proposal ids
+    confirmed_proposals: Set[str]
+
+    # Maps agreement ids to provider names
+    agreement_provider_name: Dict[str, str]
+
+    # Set of agreements confirmed by providers
+    confirmed_agreements: Set[str]
+
+    # Maps task id to task data
+    task_data: Dict[str, Any]
+
+    # Maps a provide name to the list of task ids computed by the provider
+    provider_tasks: Dict[str, List[str]]
+
+    # Map a provider name to the sum of amounts in this provider's invoices
+    provider_cost: Dict[str, float]
+
+    # Has computation finished?
+    finished: bool
+
+    def __init__(self, wrapped_emitter: Optional[Callable[[Event], None]] = None):
```

Comment:
Can you add a docstring explaining wrapped_emitter?

Reply:
Added docstring for `SummaryLogger.__init__()` in https://github.com/golemfactory/yapapi/pull/41/commits/5bc43c7f4291e6b09cd0de3c1cfef945b4f7abe9